### PR TITLE
Review before release

### PR DESCRIPTION
**Pull request by siddacious**
Had to do some git tomfoolery to set this up; I'll apply any changes for review here then apply to master on approval

My main question is about the measurement API; I got most of the unified sensor API copied over before realizing that there was no eCO2 unit in the sensor lib, so I copied the API from another eCO2 sensor.

Let me know how you'd like to proceed

### CODE REVIEW

File: Adafruit_SCD30.h [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+                                        ///< Temp data object
+  friend class Adafruit_SCD30_Humidity; ///< Gives access to private
+                                        ///< members to Humidity data
+                                        ///< object
+
+  void fillHumidityEvent(sensors_event_t *humidity, uint32_t timestamp);
+  void fillTempEvent(sensors_event_t *temp, uint32_t timestamp);
+  bool sendCommand(uint16_t command, uint16_t argument);
+  bool sendCommand(uint16_t command);
+  uint16_t getAmbiendPressure(void);
```
Comment (ladyada) [inline Adafruit_SCD30.h]: ambient

File: README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+Adafruit invests time and resources providing this open source code, please support Adafruit and open-source hardware by purchasing products from Adafruit!
+
+# Installation
+To install, use the Arduino Library Manager and search for "Adafruit SCD30" and install the library.
+
+## Dependencies
+ * [Adafruit BusIO](https://github.com/adafruit/Adafruit_BusIO)
+ * [Adafruit Unified Sensor Driver](https://github.com/adafruit/Adafruit_Sensor)
+ * [Adafruit SSD1306](https://github.com/adafruit/Adafruit_SSD1306)
+p
```
Comment (ladyada) [inline README.md]: typo?

File: Adafruit_SCD30.h [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+  bool dataReady(void);
+  bool getEvent(sensors_event_t *humidity, sensors_event_t *temp);
+
+  bool read(void);
+
+  uint16_t getMeasurementInterval(void);
+  bool setMeasurementInterval(uint16_t interval);
+
+  bool selfCalibrationEnabled(void);
+  bool selfCalibrationEnabled(bool);
```
Comment (ladyada) [inline Adafruit_SCD30.h:113]: same function name

File: examples/adafruit_scd30_test/adafruit_scd30_test.ino [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  Serial.print("Measurement Interval: "); 
+  Serial.print(scd30.getMeasurementInterval()); 
+  Serial.println(" seconds");
+}
+
+void loop() {
+  if (scd30.dataReady()){
+    Serial.println("Data available!");
+
+    if (!scd30.read()){ Serial.println("Error reading sensor data"); return; }
```
Comment (ladyada) [inline examples/adafruit_scd30_test/adafruit_scd30_test.ino]: clang-format

File: examples/adafruit_scd30_test/adafruit_scd30_test.ino [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    Serial.print("Relative Humidity: ");
+    Serial.print(scd30.relative_humidity);
+    Serial.println(" %");
+    
+    Serial.print("eCO2: ");
+    Serial.print(scd30.eCO2, 3);
+    Serial.println(" ppm");
+    Serial.println("");
+  } else {
+    //Serial.println("No data");
```
Comment (ladyada) [inline examples/adafruit_scd30_test/adafruit_scd30_test.ino]: remove

File: Adafruit_SCD30.cpp [hunk 6]
```diff
@@ -0,0 +1,516 @@
...
+
+  uint8_t buffer[5];
+  buffer[0] = (command >> 8) & 0xFF;
+  buffer[1] = command & 0xFF;
+  buffer[2] = argument >> 8;
+  buffer[3] = argument & 0xFF;
+  buffer[4] = crc8(buffer + 2, 2);
+  return i2c_dev->write(buffer, sizeof(buffer));
+}
+uint16_t Adafruit_SCD30::readRegister(uint16_t reg_address) {
```
Comment (ladyada) [inline Adafruit_SCD30.cpp:350]: add spaces between functions

**Review by ladyada — CHANGES_REQUESTED** [review]
please run clang-format on examples

c code has inconsistant spacing between functions

File: Adafruit_SCD30.h (hunk 3 above)
Comment (siddacious) [reply Adafruit_SCD30.h:113]: This was meant to be an overload, to allow it to work as a getter or setter, but I can change the API to better match the other accessors

Comment (ladyada) [reply Adafruit_SCD30.h:113]: make one `enableCalibration` and the other `isCalibrationEnabled`

**Review by ladyada — APPROVED** [review]

**Pull request merged**